test/OpenTelemetry.Collector.AspNetCore.Tests/TestCallbackMiddlewareImpl.cs

```diff
@@ -0,0 +1,32 @@
+﻿// <copyright file="TestCallbackMiddlewareImpl.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using TestApp.AspNetCore._3._1;
+
+namespace OpenTelemetry.Collector.AspNetCore.Tests
+{
+    public class TestCallbackMiddlewareImpl : CallbackMiddleware.CallbackMiddlewareImpl
```

Comment:
If I understand, this type was only used on `IncomingRequestsCollectionsIsAccordingToTheSpecTests` and was a inner class. What's the reasoning to lift this to its own file as a public class?

Reply:
We have one rule that we cant have a method and a class together. https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1201.md

I saw that if i move to the end of the file, this stops the error. will update! thanks for reviewing!

Reply:
Nice. Thanks